[BUG] Replication tests are flaky (index [books-names] blocked by: [FORBIDDEN/1000/index read-only(cross-cluster-replication))

### What is the bug?

```
OpenSearch 2.18.0


ERROR   autofollow.yaml (tests/plugins/replication/autofollow.yaml)
WARNING Multiple paths detected, please group similar tests together and move paths not being tested to prologues or epilogues.
  /_plugins/_replication/_autofollow
  /_plugins/_replication/autofollow_stats

    ERROR   EPILOGUES
        PASSED  POST /_plugins/_replication/{index}/_stop
        ERROR   DELETE /books-names (index [books-names] blocked by: [FORBIDDEN/1000/index read-only(cross-cluster-replication)];)
        PASSED  DELETE /books


ERROR   index.yaml (tests/plugins/replication/index.yaml)
WARNING Multiple paths detected, please group similar tests together and move paths not being tested to prologues or epilogues.
  /_plugins/_replication/{index}/_start
  /_plugins/_replication/{index}/_update

    ERROR   CHAPTERS
        ERROR   Start replication.
            PASSED  PARAMETERS
                PASSED  index
            PASSED  REQUEST BODY
            ERROR   RESPONSE STATUS (Expected status 200, but received 400: application/json. Cant use same index again for replication. Delete the index:books-names)
            SKIPPED RESPONSE PAYLOAD BODY
            SKIPPED RESPONSE PAYLOAD SCHEMA
            SKIPPED RESPONSE OUTPUT VALUES
        SKIPPED Update settings.
    ERROR   EPILOGUES
        ERROR   POST /_plugins/_replication/{index}/_stop (No replication in progress for index:books-names)
        PASSED  DELETE /books-names
        PASSED  DELETE /books

PASSED  stats.yaml (tests/plugins/replication/stats.yaml)
WARNING Multiple paths detected, please group similar tests together and move paths not being tested to prologues or epilogues.
  /_plugins/_replication/{index}/_status
  /_plugins/_replication/leader_stats
  /_plugins/_replication/follower_stats
  /_plugins/_replication/autofollow_stats


Tested 7/5[28](https://github.com/opensearch-project/opensearch-api-specification/actions/runs/12014616796/job/33491014455?pr=694#step:7:29) paths.
Writing coverage/test-spec-coverage-2.18.0-218a289a83d99f103c6e063e594[29](https://github.com/opensearch-project/opensearch-api-specification/actions/runs/12014616796/job/33491014455?pr=694#step:7:30)0a9879b80c1b2c08[30](https://github.com/opensearch-project/opensearch-api-specification/actions/runs/12014616796/job/33491014455?pr=694#step:7:31)15fa9752b7a36c9db.json ...
Error: Process completed with exit code 1.
```

### How can one reproduce the bug?

https://github.com/opensearch-project/opensearch-api-specification/actions/runs/12014616796/job/33491014455?pr=694

### What is the expected behavior?

Tests to always pass.

### What is your host/environment?

GHA.

### Do you have any screenshots?

![Screenshot 2024-11-25 at 11 46 40 AM](https://github.com/user-attachments/assets/aba6c3e2-57ae-47c9-9708-7907da7b3dc1)

### Do you have any additional context?

Maybe a simple retry can work around this or use `books` in one set of tests and `movies` in another?

--- a/tests/plugins/replication/autofollow.yaml
+++ b/tests/plugins/replication/autofollow.yaml
@@ -5,7 +5,7 @@ prologues:
   - path: /{index}
     method: PUT
     parameters:
-      index: books
+      index: movies
     request:
       payload: {}
   - path: /_bulk
@@ -13,14 +13,14 @@ prologues:
     request:
       content_type: application/x-ndjson
       payload:
-        - {create: {_index: books, _id: book_1392214}}
-        - {author: Harper Lee, title: To Kill a Mockingbird, year: 1960}
-        - {create: {_index: books, _id: book_1392215}}
-        - {author: Elizabeth Rudnick, title: Beauty and the Beast, year: 1991}
+        - {create: {_index: movies, _id: movie_1392214}}
+        - {director: Bennett Miller, title: Moneyball, year: 2011}
+        - {create: {_index: movies, _id: movie_1392215}}
+        - {director: Badge Dale, title: The Secret Soldiers, year: 2016}
   - path: /{index}/_refresh
     method: POST
     parameters:
-      index: books
+      index: movies
   - path: /_cluster/settings
     method: PUT
     request:
@@ -34,11 +34,11 @@ prologues:
     version: '>= 1.1'
     method: PUT
     parameters:
-      index: books-names
+      index: movies-names
     request:
       payload:
         leader_alias: leader-cluster
-        leader_index: books
+        leader_index: movies
         use_roles: 
           leader_cluster_role: all_access
           follower_cluster_role: all_access
@@ -46,13 +46,13 @@ epilogues:
   - path: /_plugins/_replication/{index}/_stop
     method: POST
     parameters: 
-      index: books-names
+      index: movies-names
     request:
       payload: {}
-  - path: /books-names
+  - path: /movies-names
     method: DELETE
     status: [200, 404]
-  - path: /books
+  - path: /movies
     method: DELETE
     status: [200, 404]
 chapters:
@@ -63,8 +63,8 @@ chapters:
     request:
       payload:
         leader_alias: leader-cluster
-        name: books
-        pattern: 'books-*'
+        name: movies
+        pattern: 'movies-*'
         use_roles:
           leader_cluster_role: leader_role
           follower_cluster_role: admin
@@ -83,6 +83,6 @@ chapters:
     request:
       payload:
         leader_alias: leader-cluster
-        name: books
+        name: movies
     response:
       status: 200

--- a/tests/plugins/replication/stats.yaml
+++ b/tests/plugins/replication/stats.yaml
@@ -5,7 +5,7 @@ prologues:
   - path: /{index}
     method: PUT
     parameters:
-      index: books
+      index: music
     request:
       payload: {}
   - path: /_bulk
@@ -13,14 +13,14 @@ prologues:
     request:
       content_type: application/x-ndjson
       payload:
-        - {create: {_index: books, _id: book_1392214}}
-        - {author: Harper Lee, title: To Kill a Mockingbird, year: 1960}
-        - {create: {_index: books, _id: book_1392215}}
-        - {author: Elizabeth Rudnick, title: Beauty and the Beast, year: 1991}
+        - {create: {_index: music, _id: music_1392214}}
+        - {author: Selena Gomez, title: Love you like a love song, year: 2011}
+        - {create: {_index: music, _id: music_1392215}}
+        - {author: Justin Bieber, title: Baby, year: 2010}
   - path: /{index}/_refresh
     method: POST
     parameters:
-      index: books
+      index: music
   - path: /_cluster/settings
     method: PUT
     request:
@@ -34,11 +34,11 @@ prologues:
     version: '>= 1.1'
     method: PUT
     parameters:
-      index: books-names
+      index: music-names
     request:
       payload:
         leader_alias: leader-cluster
-        leader_index: books
+        leader_index: music
         use_roles: 
           leader_cluster_role: all_access
           follower_cluster_role: all_access
@@ -46,13 +46,13 @@ epilogues:
   - path: /_plugins/_replication/{index}/_stop
     method: POST
     parameters: 
-      index: books-names
+      index: music-names
     request:
       payload: {}
-  - path: /books-names
+  - path: /music-names
     method: DELETE
     status: [200, 404]
-  - path: /books
+  - path: /music
     method: DELETE
     status: [200, 404]
 chapters:
@@ -61,7 +61,7 @@ chapters:
     version: '>= 1.1'
     method: GET
     parameters:
-      index: books-names
+      index: music-names
     response:
       status: 200
   - synopsis: Get leader stats.

